### PR TITLE
feat(editor): Manage poll votes

### PR DESCRIPTION
**Pull request by tomaszdurka**
- Add PollVote resolver/service
- Genearte api clients
- Introduce pagination-state-container and pagination helper component
- Add poll answer filter, allow to keep static filter data inside
- Add PollVoteList listing

### CODE REVIEW

**Comment by tomaszdurka** [comment]
@Itrulia take a look, please.

I've added some generics as I couldn't stand the way it was repeated for all these things and abused the `useEffect()`.
It's far from perfect as I made quite some compromises - would love to get some feedback.

File: libs/ui/editor/src/lib/listView/use-selectable-list.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+type UseSelectableListProps = {
+  ids?: string[]
+}
+
+export function useSelectableList({ids}: UseSelectableListProps) {
+  const [selectedItems, setSelectedItems] = useState<string[]>([])
+  const allSelected = ids && selectedItems.length === ids.length
+  const someSelected = ids && selectedItems.length > 0
+
+  const toggleAll = () => setSelectedItems(someSelected ? [] : [...(ids ?? [])])
```
Comment (Itrulia) [inline libs/ui/editor/src/lib/listView/use-selectable-list.tsx:12]:
```suggestion
  const toggleAll = () => setSelectedItems(someSelected ? [] : ids ?? []])
```

Should do the trick, no?

File: libs/ui/editor/src/lib/listView/use-selectable-list.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  const selectItem = (id: string) => {
+    setSelectedItems([...new Set([...selectedItems, id])])
+  }
+  const deselectItem = (id: string) => {
```
Comment (Itrulia) [inline libs/ui/editor/src/lib/listView/use-selectable-list.tsx:17]:
```suggestion
  const unselectItem = (id: string) => {
```

File: libs/ui/editor/src/lib/listView/use-selectable-list.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    setSelectedItems(selectedItems.filter(item => item !== id))
+  }
+  const toggleItem = (id: string | any, value?: boolean) => {
+    if (value === undefined) {
+      value = !selectedItems.includes(id)
+    }
+    if (value) {
+      selectItem(id)
+    } else {
+      deselectItem(id)
```
Comment (Itrulia) [inline libs/ui/editor/src/lib/listView/use-selectable-list.tsx:27]:
```suggestion
      unselectItem(id)
```

File: libs/ui/editor/src/lib/listView/use-selectable-list.tsx (hunk 1 above)
Comment (tomaszdurka) [reply libs/ui/editor/src/lib/listView/use-selectable-list.tsx:12]: Yeah, it should work. Was worried about reference, but it works fine. I will do my own change as there is 1 extra bracket in change 😎

**Pull request merged**